Extensions/Ansible/Src/Tasks/Ansible/ansibleCommandLineInterface.ts

```diff
@@ -0,0 +1,201 @@
+            var hostname: string = tl.getEndpointDataParameter(sshEndpoint, 'host', false);
+            var port: string = tl.getEndpointDataParameter(sshEndpoint, 'port', true); //port is optional, will use 22 as default port if not specified
+            if (!port || port === '') {
+                ansibleUtils._writeLine(tl.loc('UseDefaultPort'));
+                port = '22';
+            }
+
+            //setup the SSH connection configuration based on endpoint details
+            var sshConfig;
+            if (privateKey && privateKey !== '') {
+                tl.debug('Using private key for ssh connection.');
+                sshConfig = {
+                    host: hostname,
+                    port: port,
+                    username: username,
+                    privateKey: privateKey,
+                    passphrase: password
+                }
+            } else {
+                //use password
+                tl.debug('Using username and password for ssh connection.');
+                sshConfig = {
+                    host: hostname,
+                    port: port,
+                    username: username,
+                    password: password
+                }
+            }
+
+            var remoteCommand: string = 'ansible-playbook';
+            var playbookSource: string = tl.getInput('playbookSource', true);
+            var inventoryLocation: string = tl.getInput('inventories', true);
+            var sudoEnabled: boolean = tl.getBoolInput('sudoEnabled', false);
+            var args: string = tl.getInput('args', false);
+
+            var cleanupCmd: string[] = [];
+
+            //setup the SSH connection
+            ansibleUtils._writeLine(tl.loc('SettingUpSshConnection', sshConfig.username, sshConfig.host, sshConfig.port));
+            try {
+                sshClientConnection = await ansibleUtils.setupSshClientConnection(sshConfig);
+            } catch (err) {
+                tl.setResult(tl.TaskResult.Failed, tl.loc('ConnectionFailed', err));
+            }
+
+            if (sshClientConnection) {
+
+                if(playbookSource == 'agentMachine') {
+                    
+                    let playbookRoot: string = tl.getInput('playbookRoot', true);
+                    let playbook: string = tl.getInput('playbookPathAgentMachine', true);
+                    
+                    let remotePlaybookRoot = '/tmp/' + path.basename(playbookRoot);
+                    let remotePlaybookRootPath = '"' + remotePlaybookRoot + '"';
+                    tl.debug('ansiblePlaybookRootPath = ' + remotePlaybookRootPath);
+
+                    let scpConfig = sshConfig;
+                    scpConfig.path = remotePlaybookRoot;
+                    tl.debug('Copying playbook to ansible machine.');
+                    await ansibleUtils.copyFileToRemoteMachine(playbookRoot, scpConfig);
+
+                    let remotePlaybookPath = remotePlaybookRoot + "/" + path.basename(playbook);

```

Comment:
We cannot use path.join()
agent could be running on windows machine, in this case path would join using back slash. where as this command needs to be run on Linux which require forward slash.
If the agent is on Linux machine then we could have used path.join